README.md

```diff
@@ -22,6 +22,36 @@ fn main() {
 }
 ```
 
+# Safety
+
+It is generally not safe to concatenate two adjacent slices.
+This is explained in [#8], [bluss/odds#25], [rust-lang/rust#66111], [rust-lang/rust#62765], [rust-lang/rfcs#2806] and [tokio-rs/bytes#347].
+To sum it up, when rust calculates offsets into a slice, it makes the assumption that elements of that slice stem from the same allocation.
```

Comment:
It would be more precise to say: 

> When Rust calculates an offset to borrow part of a referenced type such as a slice, it makes the assumption that all possible offsets fall within the same allocation.

The mention of 'elements' alludes to some kind of type semantics but allocations are oblivious to validity invariants.